lib/checkclass.cpp

```diff
@@ -2402,24 +2408,24 @@ void CheckClass::checkCopyCtorAndEqOperator()
         if (!hasNonStaticVars)
             continue;
 
-        int hasCopyCtor = 0;
-        int hasAssignmentOperator = 0;
+        CtorType copyCtors = CtorType::NO;
+        CtorType assignmentOperators = CtorType::NO;
 
         std::list<Function>::const_iterator func;
         for (func = scope->functionList.begin(); func != scope->functionList.end(); ++func) {
-            if (!hasCopyCtor && func->type == Function::eCopyConstructor) {
-                hasCopyCtor = func->hasBody() ? 2 : 1;
+            if (copyCtors == CtorType::NO && func->type == Function::eCopyConstructor) {
+                copyCtors = func->hasBody() ? CtorType::WITH_BODY : CtorType::WITHOUT_BODY;
             }
-            if (!hasAssignmentOperator && func->type == Function::eOperatorEqual) {
+            if (assignmentOperators == CtorType::NO && func->type == Function::eOperatorEqual) {
                 const Variable * variable = func->getArgumentVar(0);
                 if (variable && variable->type() && variable->type()->classScope == scope) {
-                    hasAssignmentOperator = func->hasBody() ? 2 : 1;
+                    assignmentOperators = func->hasBody() ? CtorType::WITH_BODY : CtorType::WITHOUT_BODY;
                 }
             }
         }
 
-        if (std::abs(hasCopyCtor - hasAssignmentOperator) == 2)
-            copyCtorAndEqOperatorError(scope->classDef, scope->className, scope->type == Scope::eStruct, hasCopyCtor != 0);
+        if (std::abs(copyCtors - assignmentOperators) == 2)
```

Comment:
This enum calculation with std::abs is weird. I would guess it's easier to read a complete condition.